[Koh Shang Hui] iP

Prearo opened the pull request:
# Duke

> 5/7  Too much water - IGN

- Update to GUI
- Images/Resources are tentative

### How to test
1. Have [Java 11](https://www.oracle.com/java/technologies/downloads/) or higher
2. Clone the repo
3. Build in Gradle and run Launcher.java

Data is stored at`/data/duke.txt`

### Features
- [X] 3 types of tasks, Todos, Events, Deadlines
- [X] Marking by completion
- [X] Finding tasks
- [ ] Task reordering

Offline storage is currently formatted as `<type>|<mark>|<description>|[<time>]`.
This may be changed in the future to a JSON format similar to
```json
{
  "taskCount":  1
  "tasks": [
      {
        "type": "Deadline",
        "marked": 1,
        "description": "MAXXYY Project",
        "time": "2022-10-08 1800"
      }
  ]
}
```

Prearo committed:
Setup project and Level 1: Greet Echo Exit

Prearo committed:
Add, List

Prearo committed:
Mark As Done

Prearo committed:
Todo, Event and Deadline

Prearo committed:
Remove package for now

Prearo committed:
Add many exceptions

Prearo committed:
Add remove functionality

Prearo committed:
Setup automated testing input

Prearo committed:
Fix minor formatting on testing files

Prearo committed:
Separate bot driver into new class

Prearo committed:
Separate Dukebot handler into individual functions

Prearo committed:
Delete unused file messages.txt

Prearo committed:
Setup gradle and split TaskList functions.

* Add gradle with target java version 11.
* Shift add, remove, mark, unmark to a TaskList class.
* Add unused Ui and Parser classes for later revision.

Prearo committed:
Consolidate exceptions into one class

Prearo committed:
Update EXPECTED.TXT

Prearo committed:
Add storage of tasks

Prearo committed:
Add localDateTime support to Deadline

Prearo committed:
Merge branch 'branch-level-7'

Prearo committed:
Implement Ui and fix minor .gitignore errors

Prearo committed:
Fix Event and Deadline splitting regex

Prearo committed:
Implement Parser for input

Prearo committed:
Clear contents of /data/duke.txt

Prearo committed:
Revert "Clear contents of /data/duke.txt"

This reverts commit 7b1dbe5c9e8efdcdd5564c9940f89da04a036af9.

Prearo committed:
Add package

Prearo committed:
Move files to package

Prearo committed:
Add ParserTest methods

Prearo committed:
Setup shadowJar and build jar

Prearo committed:
Create JavaDocs

Prearo committed:
Update JavaDocs

Prearo committed:
Reformat code to standard

Prearo committed:
Implement find tasks in current list

Prearo committed:
Add JavaDocs for find methods

Prearo committed:
Merge branch 'branch-A-CodingStandard'

Prearo committed:
Merge branch 'branch-Level-9'

Eugene-Ong-W-X commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,69 @@
+ * Represents a list of tasks
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
```
Naming the member the same as the class might be confusing

Eugene-Ong-W-X commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,88 @@
+ * Tasks are separated by newlines in the written file
+ */
+public class Storage {
+    public final String DEFAULT_STORAGE_FILEPATH = "./data/duke.txt";
```
Only constants should be named in all caps

Eugene-Ong-W-X commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+ * Representing a deadline, i.e. a task with a specified due date
+ */
+public class Deadline extends Task {
+    protected String by;
```
by could be renamed with more details

Eugene-Ong-W-X commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,30 @@
+ * Represents an Event at a specified time
+ */
+public class Event extends Task {
+    private String at;
```
at could be renamed with more details

Eugene-Ong-W-X reviewed (commented):
Some suggestions for naming but LGTM!

Prearo committed:
Move running loop into Dukebot and rename Ui to TextUi for clarity

Prearo committed:
Fix IndexOutOfBoundsException in Storage.decode()

Prearo committed:
Add GUI functionality

Prearo committed:
Remove text-based UI remnants

Prearo committed:
Fix build.gradle entrypoint

Prearo committed:
Fix emojis not displaying in GUI

leewaichong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,220 @@
+        int index = parser.parseIndex(inputArgs);
+        taskList.remove(Integer.parseInt(inputArgs[1]) - 1);
+        StringBuilder sb = new StringBuilder();
+        sb.append(ResponseMessages.TASK_REMOVED);
```
Should this be extracted out by using a method or handler to encapsulate all these repetitive methods instead of calling it multiple times?

leewaichong commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,50 @@
+package duke.tasks;
+
+public class Task {
```
Should this be made as an abstract class since it won't be initialised?

leewaichong reviewed (commented):
Overall I like that most part of your code is encapsulated, clean and succinct.

Prearo committed:
Add Assertions in classes Duke and Storage

Storage.decode(): Under the current format for storing tasks,
the [type] of the task must be "T", "D", or "E".
The default case for the switch should never be reached.

Duke.handleInput(): An empty input after trimming is caught
by the UI level and gracefully ignored. The input here is
guaranteed to not be an empty string.

Prearo committed:
Improve code quality and handle OOB inputs

Duke: remove, mark and unmark should gracefully handle
out of bounds errors instead of exiting. A handler and new
message is added for this purpose.

MainWindow: Separated cases to ignore inputs for
future ease of editing.

Prearo committed:
Add Travis CI

Prearo committed:
Merge pull request #1 from Prearo/A-CI

Add Travis CI

Prearo committed:
Update .travis.yml

Prearo committed:
Update .travis.yml

Prearo committed:
Update .travis.yml

Prearo committed:
Update .travis.yml

Prearo committed:
Update .travis.yml

Prearo committed:
Update .travis.yml

Prearo committed:
Change build.gradle repository to non-https

Prearo committed:
Update build.gradle

Prearo committed:
Delete .travis.yml

Prearo committed:
Merge pull request #7 from Prearo/branch-A-Assertions

Add Assertions in classes Duke and Storage

Prearo committed:
Merge branch 'master' into branch-A-CodeQuality

Prearo committed:
Merge branch 'master' into branch-A-CodeQuality

Prearo committed:
Merge pull request #8 from Prearo/branch-A-CodeQuality

Improve code quality and handle OOB inputs

Prearo committed:
Add GitHub Actions

Prearo committed:
Add ability to remove multiple tasks

Prearo committed:
Merge pull request #9 from Prearo/C-MassOps

Add ability to remove multiple tasks

Prearo committed:
Fix Deadline, Event not accepting DateTime

Handling of formatted input was incorrect in Deadline.
This has been fixed with correct DateTimeFormatters.
The same extension has been applied to Event.

Prearo committed:
Merge pull request #11 from Prearo/fix-datetime

Fix Deadline, Event not accepting DateTime

Prearo committed:
Ignore data/duke.txt from VCS

Prearo committed:
Remove text-ui tests from GitHub workflow

Prearo committed:
Delete .yml

Prearo committed:
Create .yml

Prearo committed:
Merge pull request #12 from Prearo/fix-workflow

Fix workflow

Prearo committed:
Revert "Fix workflow"

Prearo committed:
Merge pull request #13 from Prearo/revert-12-fix-workflow

Revert "Fix workflow"

Prearo committed:
Remove GitHub Actions

CI has been having unknown server issues.
All workflows are unable to be viewed.

Prearo committed:
Improve GUI styling

Changes:
    Allow resize
    Change font to Consolas
    Increase font size
    Add slight background on input box

Prearo committed:
Merge pull request #14 from Prearo/A-BetterGUI

Improve GUI styling

Prearo committed:
Update README.md

Prearo committed:
Update README.md

Prearo committed:
Align output formats with README

Changed mark, unmark, remove to include
the task index in the message displayed in the UI

Prearo committed:
Update GUI

Added application image DukeMascot.png
Edited DaDuke and DaUser
Added minimum window width
Cleaned up some inconsistencies

Prearo committed:
Create and move gui to subpackage

Prearo committed:
Increase default MainWindow width

Prearo committed:
Add Ui.png

Prearo committed:
Update GUI to prevent overflow

Prearo committed:
Remove linebreaks from messages

Prearo committed:
Readd some  missing line separators

Prearo committed:
Remove unused response message
